test/foundry/unit/StakingRewardsV2/StakingRewardsV2.t.sol

```diff
@@ -782,6 +760,111 @@ contract StakingRewardsV2Test is DefaultStakingV2Setup {
     }
 
+    /*//////////////////////////////////////////////////////////////
+                        UPDATE REWARD MODIFIERS
+    //////////////////////////////////////////////////////////////*/
+
+    function test_Rewards_Updated_stake() public {
+        fundAndApproveAccountV2(address(this), TEST_VALUE);
+
+        // stake
+        stakingRewardsV2.stake(TEST_VALUE);
+
+        // configure reward rate
+        vm.prank(address(supplySchedule));
+        stakingRewardsV2.notifyRewardAmount(TEST_VALUE);
+
+        // fast forward 3 weeks
+        vm.warp(block.timestamp + 3 weeks);
+
+        // stake
+        fundAccountAndStakeV2(address(this), TEST_VALUE);
+
+        assertEq(stakingRewardsV2.lastUpdateTime(), stakingRewardsV2.lastTimeRewardApplicable());
+        assertEq(stakingRewardsV2.rewardPerTokenStored(), stakingRewardsV2.rewardPerToken());
+        assertEq(
+            stakingRewardsV2.userRewardPerTokenPaid(address(this)),
+            stakingRewardsV2.rewardPerTokenStored()
+        );
+        assertGt(stakingRewardsV2.rewards(address(this)), 0);
+    }
+
+    function test_Rewards_Updated_unstake() public {
+        fundAndApproveAccountV2(address(this), TEST_VALUE);
+
+        // stake
+        stakingRewardsV2.stake(TEST_VALUE);
+
+        // configure reward rate
+        vm.prank(address(supplySchedule));
+        stakingRewardsV2.notifyRewardAmount(TEST_VALUE);
+
+        // fast forward 3 weeks
+        vm.warp(block.timestamp + 3 weeks);
+
+        // unstake
+        unstakeFundsV2(address(this), 1);
+
+        assertEq(stakingRewardsV2.lastUpdateTime(), stakingRewardsV2.lastTimeRewardApplicable());
+        assertEq(stakingRewardsV2.rewardPerTokenStored(), stakingRewardsV2.rewardPerToken());
+        assertEq(
+            stakingRewardsV2.userRewardPerTokenPaid(address(this)),
+            stakingRewardsV2.rewardPerTokenStored()
+        );
+        assertGt(stakingRewardsV2.rewards(address(this)), 0);
+    }
+
+    function test_Rewards_Updated_stakeEscrow() public {
+        fundAndApproveAccountV2(address(this), TEST_VALUE);
+
+        // stake
+        stakingRewardsV2.stake(TEST_VALUE);
+
+        // configure reward rate
+        vm.prank(address(supplySchedule));
+        stakingRewardsV2.notifyRewardAmount(TEST_VALUE);
+
+        // fast forward 3 weeks
+        vm.warp(block.timestamp + 3 weeks);
+
+        // stake escrow
+        stakeEscrowedFundsV2(address(this), 1000);
+
+        assertEq(stakingRewardsV2.lastUpdateTime(), stakingRewardsV2.lastTimeRewardApplicable());
+        assertEq(stakingRewardsV2.rewardPerTokenStored(), stakingRewardsV2.rewardPerToken());
+        assertEq(
+            stakingRewardsV2.userRewardPerTokenPaid(address(this)),
+            stakingRewardsV2.rewardPerTokenStored()
+        );
+        assertGt(stakingRewardsV2.rewards(address(this)), 0);
+    }
+
+    function test_Rewards_Updated_unstakeEscrow() public {
```

Comment:
FWIW This is the test that broke on the synthetix codebase (and proved the bug) when you remove the modifier from unstake because it breaks the getReward calculation in exit(). Might be worth porting over

https://github.com/Synthetixio/synthetix/blob/d0070a7f70bb3ce80f6cb48d403bd45b551a7917/test/contracts/StakingRewards.js#L565-L591